### ISSUE TITLE
Review and revise logging

### ISSUE THREAD

**Issue by prjemian** [prjemian/instrument_template#2]
[Following temporary change in PR #5, added this list of requirements, including features from instrument package predecessor.]

## Logging Requirements

- [x] Configuration described in `.yml` file.
- [x] Log messages to console should be brief.
  - [x] Specify date format.
  - [x] Specify log format.
- [x] Log messages to file should be descriptive.
  - [x] Specify date format.
  - [x] Specify log format.
  - [x] Describe path to log file directory.
  - [x] Specify number of backup files to keep.
  - [x] Specify maximum size of each backup file.
- [x] Specify log level for each module.

<details>
<summary>original report</summary>

Compare the logging reports from these three branches:

### startup  from the main branch

```py
(bluesky_2024_3) $ ipython -c "from instrument.startup import *"
Activating auto-logging. Current session state plus future input saved.
Filename       : /home/prjemian/Documents/projects/prjemian/model_instrument/.logs/ipython_console.log
Mode           : rotate
Output logging : True
Raw input log  : False
Timestamping   : True
State          : active
I Wed-16:31:54 - ############################################################ startup
I Wed-16:31:54 - logging started
I Wed-16:31:54 - logging level = 10
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/_logging_setup.py
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/aps_dm_setup.py
W Wed-16:31:54 - APS DM setup file does not exist: '/home/dm/etc/dm.setup.sh'
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/debug_setup.py
Exception reporting mode: Minimal
I Wed-16:31:54 - xmode exception level: 'Minimal'
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/functions.py
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/mpl_setup.py
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/startup.py
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/best_effort.py
I Wed-16:31:54 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/catalog.py
I Wed-16:31:55 - Databroker catalog: temp
I Wed-16:31:55 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/ophyd_setup.py
I Wed-16:31:55 - using ophyd control layer: 'pyepics'
I Wed-16:31:55 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/run_engine.py
I Wed-16:31:55 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/epics_setup.py
I Wed-16:31:55 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/metadata.py
I Wed-16:31:55 - RunEngine metadata saved in directory: /home/prjemian/.config/Bluesky_RunEngine_md
I Wed-16:31:55 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/core/tests/common.py
```

### `er_test` branch

```py
(bluesky_2024_3) $ ipython -c "from instrument.startup import *"
Activating auto-logging. Current session state plus future input saved.
Filename       : /home/prjemian/Documents/projects/prjemian/model_instrument/.logs/ipython_console.log
Mode           : rotate
Output logging : True
Raw input log  : False
Timestamping   : True
State          : active
I Wed-16:35:13 - ############################################################ startup
I Wed-16:35:13 - logging started
I Wed-16:35:13 - logging level = 10
I Wed-16:35:13 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/_logging_setup.py
I Wed-16:35:13 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/aps_helper.py
I Wed-16:35:13 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/helper.py
Exception reporting mode: Minimal
I Wed-16:35:13 - xmode exception level: 'Minimal'
test 



ipython



is not notebook



W Wed-16:35:13 - APS DM setup file does not exist: '/home/dm/etc/dm.setup.sh'
I Wed-16:35:13 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/ophyd_setup.py
I Wed-16:35:13 - using ophyd control layer: 'pyepics'
```

### `er_test-specwriter-fix` branch

```py
(bluesky_2024_3) $ ipython -c "from instrument.startup import *"
Activating auto-logging. Current session state plus future input saved.
Filename       : /home/prjemian/Documents/projects/prjemian/model_instrument/.logs/ipython_console.log
Mode           : rotate
Output logging : True
Raw input log  : False
Timestamping   : True
State          : active
I Wed-16:36:08 - ############################################################ startup
I Wed-16:36:08 - logging started
I Wed-16:36:08 - logging level = 10
I Wed-16:36:08 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/_logging_setup.py
I Wed-16:36:08 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/aps_helper.py
I Wed-16:36:08 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/helper.py
Exception reporting mode: Minimal
I Wed-16:36:08 - xmode exception level: 'Minimal'
test 



ipython



is not notebook



W Wed-16:36:08 - APS DM setup file does not exist: '/home/dm/etc/dm.setup.sh'
I Wed-16:36:08 - /home/prjemian/Documents/projects/prjemian/model_instrument/src/instrument/utils/ophyd_setup.py
I Wed-16:36:08 - using ophyd control layer: 'pyepics'
```

</details>

**Comment by prjemian** [comment]
Need to review the [logging HOWTO](https://docs.python.org/3/howto/logging.html) and compare with how we use `logging` here.  The problem is that some of our our `logger.info()` statements are not producing output.  Such as:
https://github.com/prjemian/model_instrument/blob/f760bfe70f8e0389ec46558e45c80a121e9357ed/src/instrument/callbacks/spec_data_file_writer.py#L80

**Comment by prjemian** [comment]
ref: https://github.com/bluesky/bluesky/issues/1185

**Comment by MDecarabas** [comment]
commenting once more to push this higher up on priority list. This needs to get done by Monday

**Comment by prjemian** [comment]
@MDecarabas Once PR #5 is merged, including 4a7900084, I will begin work on this issue to overhaul the session logging.
